src/main.rs

```diff
@@ -16,20 +17,32 @@ struct AppState {
     pages: Arc<HashMap<String, String>>,
 }
 
+fn parse_md_to_html(markdown_input: &str) -> String {
+    let mut options = Options::empty();
+    options.insert(Options::ENABLE_STRIKETHROUGH);
+    let parser = Parser::new_ext(markdown_input, options);
+
+    let mut html_output = String::new();
+    html::push_html(&mut html_output, parser);
+    html_output
+}
+
 fn read_content(path: &Path) -> Result<HashMap<String, String>, Error> {
     let mut pages = HashMap::new();
     match read_dir(path) {
         Ok(files) => {
             for file in files {
                 let file_path = file?.path();
+                let markdown_input = &*read_to_string(&file_path)?;
+
                 pages.insert(
                     file_path
                         .file_stem()
                         .expect("Error getting filename")
                         .to_str()
                         .expect("Error converting filename to string")
                         .to_string(),
-                    read_to_string(file_path)?,
+                    parse_md_to_html(markdown_input),
```

Comment:
...and then borrow it here, taking advantage of deref coercion to provide a `&str` to the function. 

```suggestion
                    parse_md_to_html(&markdown_input),
```